pkg/webhook/admission/kfservice/kfservice_validator.go

```diff
@@ -0,0 +1,35 @@
+package kfservice
+
+import (
+	"context"
+	"net/http"
+
+	kfservingv1alpha1 "github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	admissiontypes "sigs.k8s.io/controller-runtime/pkg/webhook/admission/types"
+)
+
+// Validator that validates KFServices
+type Validator struct {
+	Client  client.Client
+	Decoder admissiontypes.Decoder
+}
+
+var _ admission.Handler = &Validator{}
+
+// Handle decodes the incoming KFService and executes Validation logic.
+func (validator *Validator) Handle(ctx context.Context, req admissiontypes.Request) admissiontypes.Response {
+	kfsvc := &kfservingv1alpha1.KFService{}
+
+	if err := validator.Decoder.Decode(req, kfsvc); err != nil {
+		return admission.ErrorResponse(http.StatusBadRequest, err)
+	}
+
+	if err := kfsvc.ValidateCreate(); err != nil {
```

Comment:
KFservice resource update will also call this method, so please rename ValidateCreate to ValidateKfservice to avoid misunderstandings.

Reply:
Eventually we may want to differentiate logic between ValidateCreate and ValidateUpdate. The Validator interface uses both of these signatures. 

Unfortunately, the current Decoder logic doesn't easily support oldObject, which is necessary for the ValidateUpdate case. 

This is written to require the most minimal code changes when upgrading to 1.13